store/tikv/region_request.go

```diff
@@ -351,17 +351,14 @@ func (s *RegionRequestSender) onRegionError(bo *Backoffer, ctx *RPCContext, seed
 		logutil.BgLogger().Debug("tikv reports `NotLeader` retry later",
 			zap.String("notLeader", notLeader.String()),
 			zap.String("ctx", ctx.String()))
-		s.regionCache.UpdateLeader(ctx.Region, notLeader.GetLeader().GetStoreId(), ctx.PeerIdx)
 
-		var boType backoffType
-		if notLeader.GetLeader() != nil {
-			boType = BoUpdateLeader
+		if notLeader.GetLeader() == nil {
+			if err = bo.Backoff(BoRegionMiss, errors.Errorf("not leader: %v, ctx: %v", notLeader, ctx)); err != nil {
+				return false, errors.Trace(err)
+			}
 		} else {
-			boType = BoRegionMiss
```

Comment:
do we have question that infinite NotLeader with Leader？

Now，backoff in TiDB has two roles:

-  sleep an interval before retry
-  control max retry time to avoid infinite retry to ensure user-end latency

it's reasonable for the first one, but do we have any situation for the second?

Reply:
It may happen when the leader's TiKV has some problems and can't handle requests forever but it is still a leader. Sounds a little weird but it's possible in some cases, e.g. bugs in grpc between TiDB and TiKV, bugs in TiKV raftstore which may cause the proposal to fail, and so on.
These cases are unnormal. Maybe we don't need to consider them.
Do you have some suggestions?

Reply:
maybe we will see other errors like `tikvrpc` to break loop, it seems that we cannot see infinite NotLeader without tikvrpc or other error - -